api/src/main/java/club/mondaylunch/gatos/api/controller/FlowController.java

```diff
@@ -372,6 +373,21 @@ public String getMetadata(
         return SerializationUtils.toJson(metadata);
     }
 
+    public record GraphErrorInfo(@JsonProperty("errors") List<GraphValidityError> errors) {
+    }
+
+    @GetMapping(value = "{flowId}/validate", produces = MediaType.APPLICATION_JSON_VALUE)
+    public String validateFlow(
+        @RequestHeader("x-user-email") String userEmail,
+        @PathVariable UUID flowId
+    ) {
+        User user = this.userRepository.getOrCreateUser(userEmail);
+        var flow = this.flowRepository.getFlow(user, flowId);
+        var graph = flow.getGraph();
+        var errors = graph.validate();
+        return SerializationUtils.toJson(new GraphErrorInfo(errors));
```

Comment:
Maybe throw an exception if there are any errors, so that the response has an error code for invalid flows?

Reply:
The resource being requested here is a list of errors, I'd say 200 is the correct thing to respond with - the request for the resource (a list of errors) succeeded.